apiserver/pkg/manager/codeflare_resource_manager.go

```diff
@@ -0,0 +1,245 @@
+	}
+	// The appwrapper is in a running state, therefore there should be a running cluster
+	// and is safe to get the state of the cluster from the kuberay api object
+	if clusterAppWrapper.Status.QueueJobState == mcadApi.AppWrapperCondDispatched &&
+		clusterAppWrapper.Status.State == mcadApi.AppWrapperStateActive {
+		return crm.DefaultResourceManager.GetCluster(ctx, clusterName, namespace)
+	}
+	return crm.converter.RayClusterFromAppWrapper(clusterAppWrapper)
+}
+
+func (crm *CodeFlareResourceManager) ListClusters(ctx context.Context, namespace string) ([]*rayv1api.RayCluster, error) {
+	rayJobAppWrapperList, err := crm.getAppWrappers(ctx, namespace)
+	if err != nil {
+		return nil, util.Wrap(err, fmt.Sprintf("List jobs failed in %s", namespace))
+	}
+	var result []*rayv1api.RayCluster
+	for _, jobAppWrapper := range rayJobAppWrapperList {
+		rayCluster, err := crm.converter.RayClusterFromAppWrapper(&jobAppWrapper)
+		if err != nil {
+			return nil, util.Wrap(err, fmt.Sprintf("Failed to extract ray cluster from app wrapper '%s/%s'", namespace, jobAppWrapper.Name))
+		}
+		result = append(result, rayCluster)
+	}
+
+	return result, nil
+}
+
+func (crm *CodeFlareResourceManager) ListAllClusters(ctx context.Context) ([]*rayv1api.RayCluster, error) {
+	namespaces, err := crm.getKubernetesNamespaceClient().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, util.Wrap(err, "Failed to fetch all Kubernetes namespaces")
+	}
+
+	var result []*rayv1api.RayCluster
+	for _, namespace := range namespaces.Items {
+		rayClusters, err := crm.ListClusters(ctx, namespace.Name)
+		if err != nil {
+			return nil, util.Wrap(err, fmt.Sprintf("List all clusters failed in %s", namespace.Name))
+		}
+		result = append(result, rayClusters...)
+	}
+	return result, nil
+}
+
+func (crm *CodeFlareResourceManager) DeleteCluster(ctx context.Context, clusterName string, namespace string) error {
+	_, err := crm.getAppWrapper(ctx, namespace, clusterName)
+	if err != nil {
+		return util.Wrap(err, "delete cluster failed")
+	}
+	return crm.mcadClient.MCADClient(namespace).Delete(ctx, clusterName, metav1.DeleteOptions{})
+}
+
+func (crm *CodeFlareResourceManager) CreateJob(ctx context.Context, apiJob *api.RayJob) (*rayv1api.RayJob, error) {
+	namespace := apiJob.Namespace
+	rayJob, err := crm.DefaultResourceManager.createRayJob(ctx, apiJob, map[string]string{
+		"ray.io/codeflare-appwrapper-name": apiJob.Name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("can't create a rayJob CRD from apiJob: %w", err)
+	}
+	jobAppWrapper, err := crm.converter.AppWrapperFromRayJob(rayJob.Get())
+	if err != nil {
+		return nil, fmt.Errorf("error creating appwrapper for ray job '%s/%s': %w", namespace, apiJob.Name, err)
+	}
+	_, err = crm.mcadClient.MCADClient(namespace).Create(ctx, jobAppWrapper, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cluster app wrapper: %w", err)
+	}
+	return rayJob.Get(), nil
```

Comment:
Similar question here as in cluster creation. What to do with potential duplicates?